checks/runchecks.go

```diff
@@ -262,15 +263,14 @@ func processTemplates(ctx *context.Context, r *pkg.CheckResult) *pkg.CheckResult
 		if tpl.IsEmpty() {
 			break
 		}
+
 		message, err := template(ctx, tpl)
 		if err != nil {
 			r.ErrorMessage(err)
-		} else if message != "true" {
-			if message != "false" {
-				r.Failf("expecting either 'true' or 'false' but got '%v'", message)
-			} else {
-				r.Failf("")
-			}
+		} else if parsed, err := strconv.ParseBool(message); err != nil {
+			r.Failf("text expression did not return a boolean value. got %s", message)
+		} else if !parsed {
+			r.Failf("test failed")
```

Comment:
Why are we adding this in ?

Reply:
without this, the check was passing for me. It's probably because we have `if Error != ""` check somewhere.

Reply:
I think we should identify that place - This will add unnecessary messages into the tables  / UI